### PR TITLE
HHH-13879 - Slow query log should use System#nanoTime not System#currentTimeMillis

### PR DESCRIPTION
**Pull request by vladmihalcea**
https://hibernate.atlassian.net/browse/HHH-13879

### CODE REVIEW

File: hibernate-core/src/main/java/org/hibernate/engine/jdbc/internal/ResultSetReturnImpl.java [hunk 1]
```diff
@@ -46,9 +46,9 @@ public ResultSetReturnImpl(JdbcCoordinator jdbcCoordinator, JdbcServices jdbcSer
 	@Override
 	public ResultSet extract(PreparedStatement statement) {
 		// IMPL NOTE : SQL logged by caller
-		long executeStart = 0;
+		long executeStartNanos= 0;
```
Comment (kortov) [inline hibernate-core/src/main/java/org/hibernate/engine/jdbc/internal/ResultSetReturnImpl.java]: I guess all the `long executeStartNanos= 0` lines here are missing a space after `executeStartNanos`

Comment (vladmihalcea) [reply hibernate-core/src/main/java/org/hibernate/engine/jdbc/internal/ResultSetReturnImpl.java]: Thanks for spotting it. I fixed it and rebased it.

Comment (NathanQingyangXu) [reply hibernate-core/src/main/java/org/hibernate/engine/jdbc/internal/ResultSetReturnImpl.java]: Saw @Sanne 's comment on your ticket. I don't know too much about it, but it seems another exciting PR from Vlad.

File: hibernate-core/src/main/java/org/hibernate/engine/jdbc/spi/SqlStatementLogger.java [hunk 2]
```diff
@@ -134,34 +135,34 @@ public void logStatement(String statement, Formatter formatter) {
...
-	public void logSlowQuery(String sql, long startTime) {
+	public void logSlowQuery(String sql, long startTimeNanos) {
 		if ( logSlowQuery < 1 ) {
 			return;
 		}
-		assert startTime > 0 : "startTime is invalid!";
-
-		long spent = System.currentTimeMillis() - startTime;
+		if ( startTimeNanos <= 0 ) {
+			throw new IllegalArgumentException( "startTimeNanos should be greater than 0!" );
```
Comment (NathanQingyangXu) [inline hibernate-core/src/main/java/org/hibernate/engine/jdbc/spi/SqlStatementLogger.java]: I would suggest providing more clue about  the exception like
```
throw new IllegalArgumentException( "startTimeNanos [" + startTimeNanos + "] should be greater than 0!" );
```
to help debugger a little bit.

**Comment by vladmihalcea** [comment]
@sebersole Thanks for merging it.

**Pull request merged**